### PR TITLE
fix: centralize compute disk limits data and update documentation

### PR DESCRIPTION
**Pull request by jordienr**
https://linear.app/supabase/issue/FE-2226/fix-incorrect-baseline-and-max-iopsthroughput-values

- Centralized compute/disk limits in packages/shared-data/compute-disk-limits.ts (MB/s, baseline+max IOPS/throughput) and re-exported from shared-data.
- Hooked Docs compute/disk table into shared data, converted docs text/table to MB/s baseline/max.
- Removed duplicated compute IOPS/throughput constants from Studio, Studio now imports shared data.
- Updated Studio disk schema: compute-aware MAX IOPS/throughput validation (MB/s), defaults computeSize to ci_micro, and gp3/io2 checks use compute caps.
- Clarified disk update mutation to send throughput as MB/s despite backend throughput_mbps field name.
- Added tests for compute-size mappings and IOPS helper logic.
- Added Infra ownership for shared compute/disk data in CODEOWNERS.
- Locked zod version via catalog and added zod dep to shared-data.

## Summary by CodeRabbit

* **New Features**
  * Added a reusable Compute Disk Limits table and centralized compute disk limits dataset for dynamic display.

* **Documentation**
  * Replaced static per-size tables with a component-driven MB/s view; clarified baseline vs. burst behavior and updated guides and troubleshooting.

* **Bug Fixes**
  * Validation and UI guidance now honor compute-size limits for IOPS and throughput.

* **Tests**
  * Expanded unit tests for sizing, mappings, and edge cases.

* **Chores**
  * Published shared-data exports, added a validation schema, pinned a dependency, and added ownership entries.

### CODE REVIEW

**Comment by jordienr** [comment]
@coderabbitai review

File: apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts [hunk 1]
```diff
@@ -159,31 +158,65 @@ export const calculateThroughputPrice = ({
...
+  price: number
+  price_interval: 'hourly' | 'monthly'
+  price_type: string
+  meta?: ProjectAddonVariantMeta
+}
 
-        const meta = option.meta as ProjectAddonVariantMeta
+type AvailableAddon = {
+  type: string
+  variants: Array<{
```
Comment (awaseem) [inline apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts:174]: Isn't this just a `ComputeAddonVariant` type, can probably replace

File: apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts [hunk 2]
```diff
@@ -159,31 +158,65 @@ export const calculateThroughputPrice = ({
...
+    meta?: unknown
+  }>
+}
 
-        return (
-          !meta.supported_cloud_providers ||
-          meta.supported_cloud_providers.includes(projectCloudProvider)
-        )
-      }) ?? []
+const isProjectAddonVariantMeta = (meta: unknown): meta is ProjectAddonVariantMeta => {
```
Comment (awaseem) [inline apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts:185]: Is there any other checks we can make. I don't think this is good enough to bound it to this object. We should see if there are some keys that exist

File: packages/shared-data/compute-disk-limits.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+export const COMPUTE_MAX_THROUGHPUT = Object.fromEntries(
+  Object.entries(COMPUTE_DISK).map(([key, value]) => [key, value.maxThroughputMBps])
+)
+
+const computeInstanceAddonVariantIds = Object.keys(COMPUTE_DISK) as ComputeDiskKey[]
+
+export const computeInstanceAddonVariantIdSchema = z.enum(
+  computeInstanceAddonVariantIds as [ComputeDiskKey, ...ComputeDiskKey[]]
+)
```
Comment (awaseem) [inline packages/shared-data/compute-disk-limits.ts:180]: I like this!

**Review by awaseem — APPROVED** [review]
Minor changes, but not blocking

File: apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts (hunk 1 above)
Comment (jordienr) [reply apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts:174]: `AvailableAddon` = raw API input, `ComputeAddonVariant` = validated output + used for manual
  construction (L221, L240)
